lusidtools/cocoon/utilities.py

```diff
@@ -392,24 +411,29 @@ def get_required_attributes_model_recursive(swagger_dict, model_object):
     return attributes
 
 
-def return_nested_model(required_attribute_properties):
+def return_nested_model(required_attribute_type):
     """
     Takes the properties of a required attribute on a model and searches as to whether or not this attribute
     requires a model of its own
 
-    :param dict required_attribute_properties: The properties of the required attribute
+    :param str required_attribute_type: The type of the required attribute
 
     :return: str: The name of the LUSID model
     """
-    possible_nested_models = list(
-        gen_dict_extract("$ref", required_attribute_properties)
-    )
 
-    if len(possible_nested_models) > 0:
-        return possible_nested_models[0].split("/")[2]
-    else:
+    if "dict" in required_attribute_type:
+        required_attribute_type = required_attribute_type.split(", ")[1].rstrip(")")
```

Comment:
what is the ',' for? is there a case when it won't be there?

Reply:
Added comments.